pylearn2/models/mlp.py

```diff
@@ -2691,6 +2691,13 @@ def get_monitoring_channels_from_state(self, state, target,
         rval = self._get_monitoring_channels_for_activations(state)
 
         return rval
+    
+    @wraps(Layer.cost)
```

Comment:
I don't think this should wrap `Layer.cost`, as `ConvNonlinearity` does not inherit from `Layer`, and they do not even have the same prototype.
The docstring you added for `IdentityConvNonlinearity`, for instance, is what should be there (with the exception of the `Notes` section).